shell/browser/electron_browser_client.cc

```diff
@@ -962,7 +962,8 @@ content::MediaObserver* ElectronBrowserClient::GetMediaObserver() {
 
 std::unique_ptr<content::DevToolsManagerDelegate>
 ElectronBrowserClient::CreateDevToolsManagerDelegate() {
-  return std::make_unique<DevToolsManagerDelegate>();
+  auto* context = ElectronBrowserContext::From("", false);
```

Comment:
Does this imply that CDP can only work for the default session?

Reply:
Yes - I'm open to other approaches but this seems closest to upstream which is why i went with it

Reply:
We definitely should be able to use CDP on non-default sessions. I'm fairly sure this is currently possible, breaking that would be a regression.

That said, it seems like this change doesn't necessarily break that? It looks like this just sets the default session as the return value from `GetDefaultBrowserContext` which is mainly used in [BrowserHandler::FindBrowserContext](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/devtools/protocol/browser_handler.cc;l=291;drc=05bb4ce2df79db9c7f4fd9ef1856f0c41bca440f) when the browser_context_id isn't specified. Which seems like exactly when we should return the default session.

Reply:
I might've misread your initial question Sam - it just implies that for this specific context of CDP usage, not CDP usage entirely. Any usage `BrowserHandler::FindBrowserContext` implies default session as jeremy noted above.